corehq/apps/accounting/management/commands/upgrade_software_plan_version.py

```diff
@@ -0,0 +1,96 @@
+import logging
+
+from django.core.management import BaseCommand
+
+from django_prbac.models import Role
+
+from corehq.apps.accounting.models import SoftwarePlanVersion
+
+logger = logging.getLogger(__name__)
+
+
+class OldRoleDoesNotExist(Exception):
+    pass
+
+
+class NewRoleDoesNotExist(Exception):
+    pass
+
+
+def upgrade_software_plan_version(old_role, new_role, limit_to_plans=None, dry_run=False):
+    """
+    :param old_role: slug for role to search for
+    :param new_role: slug for role that new software plan version should reference
+    :param limit_to_plans: optionally limit to specific plan names
+    :param dry_run: if False, will make changes to the DB
+    :return: a list of plan names that were upgraded
+    """
+    dry_run_tag = '[DRY_RUN]' if dry_run else ''
+
+    try:
+        Role.objects.get(slug=old_role)
+    except Role.DoesNotExist:
+        raise OldRoleDoesNotExist
+
+    try:
+        new_role_obj = Role.objects.get(slug=new_role)
+    except Role.DoesNotExist:
+        raise NewRoleDoesNotExist
+
+    versions = SoftwarePlanVersion.objects.filter(role__slug=old_role, is_active=True)
+    if limit_to_plans:
+        versions = versions.filter(plan__name__in=limit_to_plans)
+
+    upgraded_plans = set()
+    for old_version in versions:
```

Comment:
are you going to add a safety check to ensure that PUBLIC plans are skipped and dealt with manually?